libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1884,6 +1884,23 @@ void TypeChecker::typeCheckABIEncodeFunctions(
 			}
 		}
 
+		// The case of function call options, e.g., f{gas: 2}.
```

Comment:
Why did you not change `encodingType` instead?

Reply:
Because I told him not to :-). Change to what exactly :-)?

Reply:
Ah :) Change to return nullptr if gas or value is set?

Reply:
Yeah, ok, it came up whether to set the kind of the ``encodingType`` back to ``SetGas`` and I thought that was a bad idea and suggested this here. But you're right, it'd be better to just change ``encodingType`` to actually return ``nullptr`` for these types.

Reply:
Ok, I will change FunctionType::encodingType() to return nullptr if it has call options.